Pocket/pocket-ios
# feat(collections): create native collection view

## Summary
Create native collection view (focused on UI portion)

## References 
IN-1549

## Implementation Details
* Created native collection view in UIKit using `CollectionViewController`, `CollectionViewModel` (based on `SlateDetailViewController`, `SlateDetailViewModel`)
* For the metadata of collection, created `CollectionMetadataCell` and `CollectionMetadataPresenter` (based on ArticleMetadataPresenter)
* For the stories, created `CollectionStoryViewModel` (based on `HomeRecommendationCellViewModel`) and uses `RecommendationCell`
* Added feature flag for native collections `perm.ios.native_collections`
* Added logic to open a collection on Saves in `SavedItemsListViewModel` as well as API service `CollectionService` to retrieve graphQL data which will be refactored in the future

## Test Steps
1. Navigate to a collection in Saves list
2. Tap on the collection
3. Confirm that the native collection appears in the proper layout and format

## PR Checklist:
- N/A Added Unit / UI tests - No UI at this point of time, unit tests will be worked on in future tickets

## Screenshots
https://github.com/Pocket/pocket-ios/assets/6743397/6a5f240a-514a-4810-ab4a-7c90d4629228
